File: leet-code/938. Range Sum of BST/RangeSumOfBST938_2.java
package com.example;

public class RangeSumOfBST938_2 {
    public int rangeSumBST(TreeNode root, int low, int high) {
        if (root == null) return 0;
        int rangeSum = 0;
        if (low <= root.val && root.val <= high) rangeSum += root.val;
        rangeSum += rangeSumBST(root.left, low, high);
        rangeSum += rangeSumBST(root.right, low, high);
        return rangeSum;
    }
}

File: leet-code/938. Range Sum of BST/RangeSumOfBST938_2.kt
package com.example

class RangeSumOfBST938_2 {
    fun rangeSumBST(root: TreeNode?, low: Int, high: Int): Int {
        if (root == null) {
            return 0
        }

        if (root.`val` > high) {
            return rangeSumBST(root.left, low, high)
        }

        if (root.`val` < low) {
            return rangeSumBST(root.right, low, high)
        }

        return root.`val` + rangeSumBST(root.left, low, high) + rangeSumBST(root.right, low, high)
    }
}
